mbed_flasher/erase.py

```diff
@@ -98,61 +100,113 @@ def reset_board(self, serial_port):
         port.close()
         return EXIT_CODE_SUCCESS
 
-    def runner(self, drive):
+    def wait_to_disappear(self, mount_point):
         """
-        :param drive:
+        :param mount_point: mount_point to watch disappear
+        """
+        start_time = time()
+        while True:
+            sleep(0.1)
+            if platform.system() == 'Windows':
+                proc = Popen(["dir", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE, shell=True)
+                err = proc.stderr.read()
+            else:
+                proc = Popen(["ls", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE)
```

Comment:
Why is the shell-param not used here?

Reply:
Shell option is not needed for linux command. In windows shell=True directs the command to cmd.exe which we want to use.

Reply:
Is there a downside to using shell=True in linux? That way you could easily just select the started process array based on system outside this loop.

Reply:
https://docs.python.org/2/library/subprocess.html?highlight=subprocess%20popen#popen-constructor
The box with red content.